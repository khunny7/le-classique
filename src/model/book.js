class Book {
  constructor() {

  }

  setQuerySnapshot(querySnapshot) {
    this.querySnapshot = querySnapshot;

    this.id = this.querySnapshot.id;
    const data = this.querySnapshot.data();
    this.author = data.author;
    this.description = data.description;
    this.title = data.title;
  }
}

export { Book };