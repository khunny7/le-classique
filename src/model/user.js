class User {
  constructor(uid, data) {
    this.uid = uid;
    this.data = data || {};
  }

  setFBuser(fbUser) {
    this.fbUser = fbUser;
  }

  getData() {
    return {
      uid: this.uid,
      displayName: this.data.displayName || this.fbUser.displayName,
      photoURL: this.data.photoURL || this.fbUser.photoURL,
      email: this.data.email || this.fbUser.email,
    };
  }
}

const userConverter = {
  toFirestore: (user) => {
    const data = user.getData();

    return {
      id: data.uid,
      displayName: data.displayName,
      photoURL: data.photoURL,
      email: data.email,
    };
  },
  fromFirestore: (snapshot, options) => {
    const data = snapshot.data(options);

    return new User(snapshot.id, data);
  },
};

export { User, userConverter };
