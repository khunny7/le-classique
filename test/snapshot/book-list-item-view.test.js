import React from 'react';
import _ from 'lodash/util';
import renderer from 'react-test-renderer';
import BookListItemView from '../../src/dashboard/book-list-item-view';
import { Book } from '../../src/model/book';

jest.mock('../../src/components/cover-image', () => 'MockedCoverImage');

test('BookListItemView - ', () => {
  const book = new Book('1234', {
    author: 'Younghoon Gim',
    title: 'My story',
    cover: '',
  });
  const component = renderer.create(
    <BookListItemView
      book={book}
      onBookSelected={_.noop}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});