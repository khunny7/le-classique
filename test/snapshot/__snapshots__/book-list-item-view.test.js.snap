// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookListItemView -  1`] = `
<div
  className="book-list-item-container"
>
  <button
    className="book-list-item-btn btn btn-primary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "width": "100%",
      }
    }
    type="button"
  >
    <div
      className="book-list-item-content"
    >
      <div
        className="book-cover"
      >
        <MockedCoverImage
          coverPath=""
        />
      </div>
      <div
        className="book-detail"
      >
        <p
          className="book-title"
        >
          My story
        </p>
        <p
          className="book-author"
        >
          Younghoon Gim
        </p>
      </div>
    </div>
  </button>
</div>
`;
